libqtile/widget/keyboardlayout.py

```diff
@@ -22,107 +22,245 @@
+        (
+            'device',
+            None,
+            "Specifies the numeric device id of the input device to be updated"
+            " with the new keyboard layout. If not specified, the core"
+            " keyboard device of the X server is updated.",
+        ),
+        (
+            'display',
+            None,
+            "Specifies the display to be updated with the new keyboard layout.",
+        ),
+        (
+            'geometry',
+            None,
+            "Specifies the name of the geometry component used to construct a"
+            " keyboard layout.",
+        ),
+        (
+            'dirs',
+            None,
+            "Adds a directory to the list of directories to be used to search"
+            " for specified layout or rules files.",
+        ),
+        (
+            'keycodes',
+            None,
+            "Specifies the name of the keycodes component used to construct a"
+            " keyboard layout.",
+        ),
+        (
+            'keymap',
+            None,
+            "Specifies the name of the keymap description used to construct a"
+            " keyboard layout.",
+        ),
+        (
+            'layout',
+            None,
+            "Specifies the name of the layout used to determine the components"
+            " which make up the keyboard description. For example:"
+            " ['us', 'ru', 'sk'], 'us,ru,sk'.",
+        ),
+        (
+            'model',
+            None,
+            "Specifies the name of the keyboard model used to determine the"
+            " components which make up the keyboard description. For"
+            " example: 'pc101', 'pc105'.",
+        ),
+        (
+            'option',
+            None,
+            "Specifies the name of an option to determine the components which"
+            " make up the keyboard description. For example:"
+            " ['compose:menu', 'grp_led:scroll'],"
+            " 'compose:menu,grp_led:scroll'.",
+        ),
+        (
+            'rules',
+            None,
+            "Specifies the name of the rules file used to resolve the"
+            " requested layout and model to a set of component names.",
+        ),
+        (
+            'symbols',
+            None,
+            "Specifies the name of the symbols component used to construct a"
+            " keyboard layout.",
+        ),
+        (
+            'synch',
+            False,
+            "Force synchronization for X requests.",
+        ),
+        (
+            'types',
+            None,
+            "Specifies the name of the types component used to construct a"
+            " keyboard layout.",
+        ),
+        (
+            'variant',
+            None,
+            "Specifies which variant of the keyboard layout should be used to"
+            " determine the components which make up the keyboard"
+            " description.",
+        ),
     ]
+    dependencies = {
+        'setxkbmap': ['setxkbmap', '-version'],
```

Comment:
Can we depend on either xkb-switch or setxkbmap, but not both?

Reply:
Yes, we can make setxkbmap an optional dependency (then the user will have to configure it via `/etc/X11/xorg.conf.d/00-keyboard.conf`).